internal/concierge/server/prepare_controllers.go

```diff
@@ -164,6 +162,7 @@ func PrepareControllers(c *Config) (controllerinit.RunnerBuilder, error) { //nol
 			apicerts.NewAPIServiceUpdaterController(
 				c.ServerInstallationInfo.Namespace,
 				c.NamesConfig.ServingCertificateSecret,
+				apicerts.RetrieveCAFromSecret,
```

Comment:
Not sure I understand why the controllers need these new dependencies injected?

Reply:
They are exported, global functions, so theoretically they could just be called inline.

The reason they are injected as dependencies is just so the unit tests of these controllers can provide their own implementation of this function, which means that it does not need to be aware of the specific keys used in the secret (it can make up its own keys).